plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftPublishController.java

```diff
@@ -108,6 +111,8 @@ public void publishFinish(IProgressMonitor monitor) throws CoreException {
 		super.publishFinish(monitor);
 
 		final File localFolder = getDeploymentsRootFolder();
+		syncPodsToDirectory(localFolder, monitor);
```

Comment:
Why is syncPodsToDirectory being done during publishFinish?  Shouldn't that be done at the start of the publish operation, like publishStart or something? My concern is if this is done after publish, it may be buggy.

For example:
   User changes Sample.war.  publishStart() is called, publishModule(etc) is called.  You copy in the new changes of Sample.war,  then you syncPodsToDirectory and end up overwriting your changes? Not sure if this is correct but its my current concern.

Reply:
Fair point but I dont think that this matters given that the local files are newer and thus wont be overwritten. Still, I'll move sync down to publishing start given that it's more logical.